spec/models/nonprofit_spec.rb

```diff
@@ -121,7 +124,11 @@
         expect(nonprofit.errors['state_code'].first).to match /.*blank.*/
       end
 
-      it 'rejects an invalid user' do 
+      it 'has an error for not in the US state' do
+        expect(nonprofit.errors['state_code']).to include 'must be a US two-letter state code'
+      end
```

Comment:
Make sure that no error is added for a state_code if state_code is set to a non-capitalized version of the state_code. As an example, if you set state_code to "Wi" then the validation should pass.

Reply:
Okay, it's a good test case, we'll add it!

While investigating it I've found another problem. We created a `nonprofit_with_not_US_state` but we didn't use it in any test case, a new test case should also be added to cover this case.

Changing the validation method to validate the state code only when it's not nil (as mentioned in another comment), it could look like this:

```rb
  let(:nonprofit_with_not_US_state) { Nonprofit.new(user_id: user.id, state_code: 'KK')}

  it 'has an error for not in the US state' do
     expect(nonprofit_with_not_US_state.errors['state_code']).to match_array ['must be a US two-letter state code']
  end
```

Because it should only have a single error.